zerver/tests/test_subs.py

```diff
@@ -3120,9 +3126,11 @@ def _test_user_settings_for_creating_streams(
 
         do_set_realm_property(realm, stream_policy, Realm.POLICY_ADMINS_ONLY, acting_user=None)
         do_change_user_role(user_profile, UserProfile.ROLE_MODERATOR, acting_user=None)
+        # When this function is run a second time for public stream test, the streams already exists.
+        # So, create a new name.
```

Comment:
This is a messy fix; for this class at least, we should just have independent `test_foo` functions that call the `_test_` method, since then they live in different transactions, so streams created in one don't affect the other.

And maybe that suggests doing that migration for all of them in this test class.

Reply:
Done